crates/core/src/client/config.rs

```diff
@@ -0,0 +1,37 @@
+use eyre::{eyre, Result};
+use starknet::core::types::FieldElement;
+
+fn get_env_var(name: &str) -> Result<String> {
+    std::env::var(name).map_err(|_| eyre!("Missing mandatory environment variable: {name}"))
+}
+
+pub struct StarknetConfig {
+    pub starknet_rpc: String,
+    pub kakarot_address: FieldElement,
+    pub proxy_account_class_hash: FieldElement,
+}
+
+impl StarknetConfig {
+    pub fn new(
+        starknet_rpc: &str,
+        kakarot_address: FieldElement,
+        proxy_account_class_hash: FieldElement,
+    ) -> StarknetConfig {
```

Comment:
this can be `self`
funny that clippy didnt catch it

Reply:
Oh! Makes sense!

Yeah, something that should have been caught by Clippy!